### ISSUE TITLE
Allow lazy configuration of svgs through the url function

### ISSUE THREAD

**Issue by ChristopherPHolder** [push-based/ngx-fast-svg#77]
## TL;DR

As a user i want to be able to fetch the configuration for my svgs and not require it to be present at bootstrap

## Description

Currently the configuration for the url for the svgs need to be configured or available at bootstrap. However for large application, specially multitenant applications, its necessary to configure more reactively. 

It is currently possible to load the configuration before the app has bootstrapped but this might delay the bootstrapping or load request before they are necessary. It would also be possible to get the configuration before the component has bootstrap but this would mean we do not have any placeholders before that.

Ideally, we would like to be able to bootstrap the component before the configuration has loaded. After this configuration has loaded we would then react to by initializing out loader. 

The reason this is currently not possible is the `url` function returns a string and is not reactive:

```ts
export type SvgOptions = {
  url: (name: string) => string;
  defaultSize?: string;
  suspenseSvgString?: string;
};
```

Then this url is used inside of an `untracked` function inside an `effect` of the `FastSvgComponent` to initilize the loader. 

```ts
effect(() => {
  untracked(() => {
    if (!name) {
      throw new Error('svg component needs a name to operate');
    }
  
    if (!this.registry.isSvgCached(name)) {

      // Here we run the url function and added to an image tag
      // This is used to piggy back on the img lazy loading and allow the browser to decide when it should load the svg
      const i = this.getImg(this.registry.url(name)); 
      this.renderer.appendChild(this.element.nativeElement, i);
      img = elem.querySelector('img') as HTMLImageElement;
      addEventListener(img, 'load', this.loadedListener);

    }
  });
});
```

However consider the what happens if the config that provides the url has not loaded yet.

We do not handle passing it an undefined value and we do not handle reacting to a value change.

## Potential solution

Handle url reactivity by an observable of a string instead of for the url tranformer.

## TODOs
- [x] Make the url reactive
- [x] Handle lazy url
- [x] Ensure no side-effects or unnecessary change detection
- [x] Fix the docs to reflect changes
